oop-relationships/practice-inheritance.md

```diff
@@ -0,0 +1,7 @@
+# Practice: Inheritance
+
+## Directions
+
+Complete all questions below.
+
+## Practice
```

Comment:
Here's what I've got so far:

<!-- prettier-ignore-start -->
### !challenge
* type: checkbox
* id: 85c8a657-1907-418a-9020-7b7baf02b9d1
* title: Inheritance practice I

##### !question
What is the key difference between inheritance and composition ?
##### !end-question

##### !options
* Composition is an extension of a parent class, while Inheritance is more of a distant cousin twice removed. 
* Inheritance describes a `is-A` relationship while composition describes `has-A` relationship. 
* Inheritance can be described as an extension of a parent class, while composition adds supportive logic to a class. 
##### !end-options

##### !answer
* Inheritance describes a `is-A` relationship while composition describes `has-A` relationship. 
* Inheritance can be described as an extension of a parent class, while composition adds supportive logic to a class. 
##### !end-answer
### !end-challenge
<!-- prettier-ignore-end -->


<!-- prettier-ignore-start -->
### !challenge
* type: checkbox
* id: 9247804e-3ef4-4995-b285-693d74dfdd04
* title: Inheritance Practice II

##### !question
Provided this code snippet, select the option that best describes how inheritance works between the `Bear` and `Mammal` class.

``` Python
class Mammal:
    def __init__(self, name, region):
        self.name = name
        self.region = region
    
    def make_noise():
        print("AHHHHHHH")

class Bear(Mammal):
    def __init__(self, color, size):
        self.color = color
        self.size = size

    def make_appearance():
        print("ʕ •ᴥ• ʔ")
```
##### !end-question

##### !options
* `Bear` will inherit `name`, `region`, and `make_appearance()` from `Mammal`.
* `Mammal` will inherit `color`, `size`, and `make_appearance()` from `Mammal`.
* `Bear` will inherit `name`, `region`, and `make_noise()` from `Mammal`.
##### !end-options

##### !answer
* `Bear` will inherit `name`, `region`, and `make_noise()` from `Mammal`.
##### !end-answer
### !end-challenge
<!-- prettier-ignore-end -->

<!-- prettier-ignore-start -->
### !challenge
* type: checkbox
* id: d48131b6-61c3-4385-8dbc-ffb41103980f
* title: Inheritance III

##### !question
Petunia's Petshop is creating a program to keep track of pet inventory. Using inheritance, select the option that generates the following output:

```
There are 2 tabby cats in stock.
They each cost $200.
Here is what the cat(s) looks like:

=^._.^= ∫ 
=^._.^= ∫

```
##### !end-question

##### !options
* 
``` Python
class Cat:
    def __init__(self, name, image):
        self.name = name
        self.image = image

    def display_picture(self, image):
        print(image)

class Cat():
    def __init__(self, breed, price, count):
        self.breed = breed
        self.price = price
        self.count = count

    def display_summary(self):
        print(f"There are {self.count} cats in stock.")
        for i in range(self.count):
            Pet.display_picture("name",
            "\n=^._.^= ∫")

tabby = Cat("tabby", 20, 10) 
tabby.display_summary()       
```

* 
``` Python
class Pet:
    def __init__(self, name, image):
        self.name = name
        self.image = image

    def display_picture(self, image):
        print(image)

class Cat(Pet):
    def __init__(self, breed, price, count):
        self.breed = breed
        self.price = price
        self.count = count

    def display_summary(self):
        print(f"There are {self.count} cats in stock.")

tabby = Cat("tabby", 20, 10) 
tabby.display_summary()       
```
* 
``` Python
class Pet:
    def __init__(self, name, image):
        self.name = name
        self.image = image

    def display_picture(self, image):
        print(image)

class Cat(Pet):
    def __init__(self, breed, price, count):
        self.breed = breed
        self.price = price
        self.count = count

    def display_summary(self):
        print(f"There are {self.count} cats in stock.")
        for i in range(self.count):
            Pet.display_picture("name",
            "\n=^._.^= ∫")

tabby = Cat("tabby", 20, 10) 
tabby.display_summary()       
```
##### !end-options

##### !answer
* 
``` Python
class Pet:
    def __init__(self, name, image):
        self.name = name
        self.image = image

    def display_picture(self, image):
        print(image)

class Cat(Pet):
    def __init__(self, breed, price, count):
        self.breed = breed
        self.price = price
        self.count = count

    def display_summary(self):
        print(f"There are {self.count} cats in stock.")
        for i in range(self.count):
            Pet.display_picture("name",
            "\n=^._.^= ∫")

tabby = Cat("tabby", 20, 10) 
tabby.display_summary()       
```
##### !end-answer
### !end-challenge
<!-- prettier-ignore-end -->